### DISCUSSION TITLE
BaramMesh Loas Mesh

### DISCUSSION THREAD

**Discussion by torodecamargue** [nextfoam/baram#90] [General]
Hello Baram team,

I use Baram since few weeks now and I think to an improvement in term of saving time during meshing operations.
After each change in operations (base grid, castellation, snap, BND layer) the mesh is reloaded and it takes a very long time to load with billion cells.
Do you think, adding a button asking the user to load or not would be possible ? 
I think it would save a lot of time, for example when setting the last details in the mesh we don't need to control each steps the mesh visualisation.
It would be more flexible, the user could choose and save time.

![image](https://github.com/nextfoam/baram/assets/75060318/87f9dabc-f9d1-402b-935d-01fc77e7a0fd)

Best regards

Julien

**Comment by jiban** [comment]
Hi, Julien.
Thank you for the suggestion.
I agree that it takes long time to load a mesh especially when the mesh is big.
However, it would also be confusing if the parameters on left-side do not coincide with the mesh on the right-side.
Could you elaborate on telling me the needs in detail?
In which situation you don't want to reload the mesh even though you changes some parameters?
Or do you just need to see/check the parameters you've configured?

-- Jake

**Comment by torodecamargue** [comment]
Hello Jake,

I think that you can have a button "update mesh" telling that the display of the mesh is updated with a green arrow for example or a red cross. The user will have to do an action, if he needs, to update the display of the mesh.

In my personnal case, when I want to update the mesh with new parameters, I am only interested in the final display after boundary layer insertion. I don't need to see the mesh after the base grid or snap operations because during the first trial I have already check that it was ok.
When we want to check the parameters I have to relaod the mesh at each operations... I think a button to load the mesh should give more flexibility.

The mesh loading is approx 25% of my time when meshing biges cases, I think it is too long and I don't need it 90% of the time.

Best regards 

Julien

**Comment by jiban** [comment]
Hi Julien,
Thank you for sharing your valuable thoughts.
Now I could fully understand your needs and agree that it's quite a necessary feature.
I will put this feature request into higher priority queue.

Best regards,
Jake

**Comment by jiban** [comment]
Hi Julien,
I'm happy to let you know that the feature has been implemented in v24.4.0.
You can turn off the update by toggling the button below.

![image](https://github.com/user-attachments/assets/823b040d-bb91-4f4c-9d82-d3054be09118)

Please try it and let me know if it's ok.
Thank you.
